pkg/contract-manager/cr_services.go

```diff
@@ -0,0 +1,151 @@
+package contractmanager
+
+import (
+	"context"
+
+	nodecorev1alpha1 "fluidos.eu/node/api/nodecore/v1alpha1"
+	reservationv1alpha1 "fluidos.eu/node/api/reservation/v1alpha1"
+	discoverymanager "fluidos.eu/node/pkg/discovery-manager"
+	localResourceManager "fluidos.eu/node/pkg/local-resource-manager"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// getAllFlavours returns all the Flavours in the cluster
+func getAllFlavours(cl client.Client) ([]nodecorev1alpha1.Flavour, error) {
+
+	var flavourList nodecorev1alpha1.FlavourList
+
+	// List all Flavour CRs
+	err := cl.List(context.Background(), &flavourList)
+	if err != nil {
+		return nil, err
+	}
+
+	return flavourList.Items, nil
+}
+
+// getFlavourByID returns the Flavour CR in the cluster that matches the flavourID
+func getFlavourByID(flavourID string, cl client.Client) (*localResourceManager.Flavour, error) {
+	flavours, err := getAllFlavours(cl)
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavourCR := range flavours {
+		if flavourCR.Name == flavourID {
+			// Return from the Flavour CR the Flavour struct
+			flavour := parseCRToFlavour(flavourCR)
+			return flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// getFlavourByIDCR returns the entire Flavour CR (not only spec) in the cluster that matches the flavourID
+func getFlavourByIDCR(flavourID string, cl client.Client) (*nodecorev1alpha1.Flavour, error) {
+	flavoursSpec, err := getAllFlavours(cl)
+	if err != nil {
+		klog.Errorf("Error getting all the Flavour CRs: %s", err)
+		return nil, err
+	}
+
+	// Get the Flavour that matches the flavourID
+	for _, flavour := range flavoursSpec {
+		if flavour.Name == flavourID {
+			return &flavour, nil
+		}
+	}
+
+	return nil, nil
+}
+
+// filterFlavoursBySelector returns the Flavour CRs in the cluster that match the selector
+func filterFlavoursBySelector(flavours []localResourceManager.Flavour, selector Selector) ([]localResourceManager.Flavour, error) {
+	var flavoursSelected []localResourceManager.Flavour
+
+	// Get the Flavours that match the selector
+	for _, f := range flavours {
+		if f.Type == selector.FlavourType {
+			if f.Characteristics.CPU >= selector.CPU && f.Characteristics.RAM >= selector.RAM {
+				flavoursSelected = append(flavoursSelected, f)
+			}
+		}
+	}
+
+	return flavoursSelected, nil
+}
```

Comment:
This is not the function that we talked about yesterday!

- We decided to filter on `[]nodecorev1alpha1.Flavour` and not on `[]localResourceManager.Flavour`
- We said that this should be a more complex function of filtering, take inspiration (as we said yesterday) from the `filterPeeringCandidate` --> `/pkg/rear/common.go`

Reply:
Now I used a more complex function (following the actual REAR logic for selector).